internal/service/location/key.go

```diff
@@ -0,0 +1,372 @@
+				Computed: true,
+				Optional: true,
+				Validators: []validator.String{
+					stringvalidator.LengthBetween(0, 1000),
+				},
+				Default: stringdefault.StaticString("Managed By Terraform"),
+			},
+			"expire_time": schema.StringAttribute{
+				Optional: true,
+			},
+			"force_update": schema.BoolAttribute{
+				Optional: true,
+			},
+			"id": framework.IDAttribute(),
+			"key": schema.StringAttribute{
+				Computed: true,
+				PlanModifiers: []planmodifier.String{
+					stringplanmodifier.UseStateForUnknown(),
+				},
+			},
+			"key_arn": schema.StringAttribute{
+				CustomType: fwtypes.ARNType,
+				Computed:   true,
+				PlanModifiers: []planmodifier.String{
+					stringplanmodifier.UseStateForUnknown(),
+				},
+			},
+			"key_name": schema.StringAttribute{
+				Required: true,
+				Validators: []validator.String{
+					stringvalidator.LengthBetween(1, 100),
+				},
+				PlanModifiers: []planmodifier.String{
+					stringplanmodifier.RequiresReplace(),
+				},
+			},
+			"no_expiry": schema.BoolAttribute{
+				Optional: true,
+			},
+			"update_time": schema.StringAttribute{
+				Computed: true,
+				PlanModifiers: []planmodifier.String{
+					stringplanmodifier.UseStateForUnknown(),
+				},
+			},
+			names.AttrTags:    tftags.TagsAttribute(),
+			names.AttrTagsAll: tftags.TagsAttributeComputedOnly(),
+		},
+		Blocks: map[string]schema.Block{
+			"restrictions": schema.ListNestedBlock{
+				CustomType: fwtypes.NewListNestedObjectTypeOf[restrictionsData](ctx),
+				Validators: []validator.List{
+					listvalidator.SizeAtMost(1),
+					listvalidator.IsRequired(),
+				},
+				NestedObject: schema.NestedBlockObject{
+					Attributes: map[string]schema.Attribute{
+						"allow_actions": schema.ListAttribute{
+							ElementType: types.StringType,
+							Required:    true,
+						},
+						"allow_referers": schema.ListAttribute{
+							Computed: true,
+							Default: listdefault.StaticValue(
+								types.ListValueMust(types.StringType, []attr.Value{types.StringValue("*")}),
+							),
+							ElementType: types.StringType,
+							Optional:    true,
+						},
+						"allow_resources": schema.ListAttribute{
+							ElementType: types.StringType,
+							Required:    true,
+						},
+					},
+				},
+			},
+		},
+	}
+}
+
+func (r *resourceKey) Create(ctx context.Context, req resource.CreateRequest, resp *resource.CreateResponse) {
+	conn := r.Meta().LocationClient(ctx)
+
+	var plan resourceKeyData
+	resp.Diagnostics.Append(req.Plan.Get(ctx, &plan)...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	in := &location.CreateKeyInput{}
+	resp.Diagnostics.Append(flex.Expand(ctx, plan, in)...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	// 'ExpireTime' must not be set when 'NoExpiry' has value true.
+	if !plan.NoExpiry.IsNull() {
+		in.NoExpiry = aws.Bool(plan.NoExpiry.ValueBool())
+	} else if !plan.ExpireTime.IsNull() {
+		expireTime, _ := time.Parse(time.RFC3339, plan.ExpireTime.ValueString())
```

Comment:
Copied precedent from https://github.com/hashicorp/terraform-provider-aws/blob/6e919e2821c68a62cd312e44ae7ce958419f23db/internal/service/appsync/api_key.go#L76 for time parsing.